test/pr_test.go

```diff
@@ -126,31 +94,17 @@ func TestRegistrationToCtxReleaseWithPDUSession(t *testing.T) {
 	}
 
 	time.Sleep(time.Duration(5000) * time.Millisecond)
-	assert.Equalf(t, ueCount, createdSessionCount, "Expected %d PDU sessions created but was %d", ueCount, createdSessionCount)
-	assert.Equalf(t, ueCount, releasedSessionCount, "Expected %d PDU sessions created but was %d", ueCount, releasedSessionCount)
-	assert.Equalf(t, ueCount, releasedCtxCount, "Expected %d PDU sessions created but was %d", ueCount, releasedCtxCount)
+	fiveGC.GetAMFContext().ExecuteForAllUe(
```

Comment:
very cool mechanism!
we need to make sure it went through the other states as well
Adding some AssertJ could make thing even better